repository/github_user.go

```diff
@@ -0,0 +1,96 @@
+//Package repository container the fuctions to handle githubusers model information
+package repository
+
+import (
+	"encoding/csv"
+	"io"
+	"log"
+	"lolidelgado/github-users/models"
+	"os"
+	"strconv"
+)
+
+type GithubUser struct {
+	githubUserCsvFileName string
+}
+
+type IGithubUserRepository interface {
+	FetchAll() ([]models.GithubUser, error)
+}
+
+const staticFilesPath = "static/"
+
+func NewGithubUser(fileName string) *GithubUser {
+	return &GithubUser{
+		fileName,
+	}
+}
+
+func (g *GithubUser) FetchAll() ([]models.GithubUser, error) {
+	lines, err := g.readCsv()
+	if err != nil {
+		return nil, err
+	}
+	return arrayToGithubUser(lines), nil
+}
+
+func (g *GithubUser) GetById(id int) ([]models.GithubUser, error) {
+	lines, err := g.readCsv()
+	if err != nil {
+		return nil, err
+	}
+	var user [][]string
+	for _, line := range lines {
+		if idFromFile, _ := strconv.Atoi(line[0]); idFromFile == id {
+			user = append(user, line)
+			break
+		}
+	}
+	if len(user) > 0 {
+		return arrayToGithubUser(user), nil
+	} else {
+		return []models.GithubUser{}, nil
+	}
+}
+
+func (g *GithubUser) readCsv() ([][]string, error) {
+	//csv reader
+	file, err := os.Open(staticFilesPath + g.githubUserCsvFileName)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+	csvReader := csv.NewReader(file)
+
+	var lines [][]string
+	for {
+		record, err := csvReader.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			log.Fatal(err)
+		}
+		if _, err := strconv.Atoi(record[0]); err == nil {
+			lines = append(lines, record)
+		}
+	}
+	return lines, nil
+}
+
+func arrayToGithubUser(lines [][]string) []models.GithubUser {
+	var githubUsers []models.GithubUser
+	for _, line := range lines {
+		id, err := strconv.Atoi(line[0])
+		if err != nil {
+			continue
+		}
+		publicRepos, err := strconv.Atoi(line[5])
+		if err != nil {
+			publicRepos = 0
+		}
+		data := models.NewGithubUser(id, line[1], line[2], line[3], line[4], publicRepos)
+		githubUsers = append(githubUsers, data)
+	}
+	return githubUsers
+}
```

Comment:
Nitpicking here, add spaces for better readability.
```suggestion 
func arrayToGithubUser(lines [][]string) []models.GithubUser {
	var githubUsers []models.GithubUser
	
	for _, line := range lines {
		id, err := strconv.Atoi(line[0])
		if err != nil {
			continue
		}
		
		publicRepos, err := strconv.Atoi(line[5])
		if err != nil {
			publicRepos = 0
		}
		
		data := models.NewGithubUser(id, line[1], line[2], line[3], line[4], publicRepos)
		githubUsers = append(githubUsers, data)
	}
	
	return githubUsers
}
```